pkg/reconciler/buildrun/resources/sources/git.go

```diff
@@ -35,32 +52,33 @@ func AppendGitStep(
 	gitStep.Container.Name = fmt.Sprintf("source-%s", name)
 	gitStep.Container.Args = []string{
 		"--url",
-		source.URL,
+		url,
 		"--target",
-		fmt.Sprintf("$(params.%s-%s)", prefixParamsResultsVolumes, paramSourceRoot),
+		filepath.Join(fmt.Sprintf("$(params.%s-%s)", prefixParamsResultsVolumes, paramSourceRoot),
```

Comment:
I prefer to run `path.Join()` here like in https://github.com/shipwright-io/build/blob/328f8c035a23a908fee1623d91947a1b0767bf51/pkg/reconciler/buildrun/resources/taskrun.go#L304-L310 to make sure that the user input for the path is commonly applied and sanitized (double-slashes, ../, etc).